raiden-ts/src/raiden.ts

```diff
@@ -1043,14 +1044,21 @@ export class Raiden {
     // Check whether we are on a test network
     assert(this.deps.network.chainId !== 1, ErrorCodes.RDN_MINT_MAINNET, this.log.info);
 
-    const { signer } = chooseOnchainAccount(this.deps, subkey ?? this.config.subkey);
+    const { signer, address } = chooseOnchainAccount(this.deps, subkey ?? this.config.subkey);
     // Mint token
     const customTokenContract = CustomTokenFactory.connect(token, signer);
 
+    const beneficiary = to ?? address;
+    assert(
+      Address.is(beneficiary),
+      [ErrorCodes.DTA_INVALID_ADDRESS, { beneficiary }],
+      this.log.info,
+    );
+
     const receipt = await callAndWaitMined(
       customTokenContract,
-      'mint',
-      [decode(UInt(32), amount)],
+      'mintFor',
+      [decode(UInt(32), amount, ErrorCodes.DTA_INVALID_AMOUNT), beneficiary],
```

Comment:
Please pull out the decoding. Must not be mixed in here. I think we don't do so elsewhere. Anyways I would say this is a bad habit. :wink: